src/main/java/dk/cngroup/lentils/service/ScoreService.java

```diff
@@ -51,4 +54,17 @@ public int getScoreByTeam(final Team team) {
                 .collect(Collectors.toList());
         return teamScores;
     }
+
+    public List<TeamScoreDetail> getTeamWithDetailScores(final Team team) {
+        List<TeamScoreDetail> teamScoreDetails = new ArrayList<>();
+        final List<Cypher> cyphers = cypherService.getAll();
+        for (Cypher cypher: cyphers) {
```

Comment:
OPTIONAL: Kdyz uz je tam for cyklus, tak bych asi zvolil tento zapis:
```java
        for (Cypher cypher: cyphers) {
            TeamScoreDetail teamScoreDetail = new TeamScoreDetail(
                    cypher,
                    statusService.getStatusScore(team, cypher),
                    hintTakenService.getAllByTeamAndCypher(team, cypher),
                    hintTakenService.getHintScore(team, cypher),
                    this.getScoreByTeamAndCypher(team, cypher));
            teamScoreDetails.add(teamScoreDetail);
        }
```
Z duvodu prehlednosti, protoze neni `teamScoreDetails.add `a new `TeamScoreDetail` na jednom radku a nejsou tam jako kdyby dve zacinajici zanoreni.

Jinak jen pro zajimavost by sel pouzit ten stream, ktery jsi rikala, ze ti nesel nejak takto (ale asi ho tam nedavaj. Dulezitejsi je, abyste tomu rozumneli vy a ke streamum se dostaneme ;) ):
```java
cyphers.stream()
            .map(cypher -> new TeamScoreDetail(
                        cypher,
                        statusService.getStatusScore(team, cypher),
                        hintTakenService.getAllByTeamAndCypher(team, cypher),
                        hintTakenService.getHintScore(team, cypher),
                        this.getScoreByTeamAndCypher(team, cypher))))
            .collect(Collectors.toList())
```